Source/LibationWinForms/grid/ProductsGrid.cs

```diff
@@ -62,14 +65,37 @@ private async void DataGridView_CellContentClick(object sender, DataGridViewCell
 			if (e.RowIndex < 0)
 				return;
 
-			var clickedColumn = _dataGridView.Columns[e.ColumnIndex];
-
-			if (clickedColumn == liberateGVColumn)
+			if (e.ColumnIndex == liberateGVColumn.Index)
 				await Liberate_Click(getGridEntry(e.RowIndex));
-			else if (clickedColumn == tagAndDetailsGVColumn)
+			else if (e.ColumnIndex == tagAndDetailsGVColumn.Index)
 				Details_Click(getGridEntry(e.RowIndex));
-			else if (clickedColumn == descriptionGVColumn)
+			else if (e.ColumnIndex == descriptionGVColumn.Index)
 				Description_Click(getGridEntry(e.RowIndex), _dataGridView.GetCellDisplayRectangle(e.ColumnIndex, e.RowIndex, false));
+			else if (e.ColumnIndex == coverGVColumn.Index)
+				await Cover_Click(getGridEntry(e.RowIndex));
+		}
+
+		private async Task Cover_Click(GridEntry liveGridEntry)
+		{
+			var picDefinition = new PictureDefinition(liveGridEntry.LibraryBook.Book.PictureLarge, PictureSize.Native);
+			var picDlTask = Task.Run(() => PictureStorage.GetPictureSynchronously(picDefinition));
```

Comment:
Does this need an await ?

Reply:
Does this need `await Task.Run`... ?

(Forgive me if this is repetitive, I'm fairly unfamiliar with how github does things for collaboration. I'm used to working solo on github or in a team not on github)